tests/test_gh.py

```diff
@@ -69,131 +68,163 @@ def test_get_issue_content_bad(mocker):
+        # C3: a directory input and YAML files exist in the directory.
+        #   Expect the broadcast repos dict is returned.
+        (
+            "repo_info_dir_yaml",
+            "dir",
+            None,
+            None,
+            "default",
+        ),
+        # C4: no input, cwd have `repos.json` and `groups.json`,
+        #   `~/.skpkgrc` have `url_to_repo_info`.
+        #   Expect the broadcast repos dict specified in the cwd is returned.
+        (
+            None,
+            "none",
+            "repo_info_dir_json",
+            "home_dir",
+            "default",
+        ),
+        # C5: no input, cwd does not have the files.
+        #   `~/.skpkgrc` have `url_to_repo_info`.
+        #   Expect the broadcast repos dict specified in the
+        #   `url_to_repo_info` is returned.
+        (
+            None,
+            "none",
+            "empty-source-dir",
+            "home_dir",
+            "another",
+        ),
+    ],
+)
+def test_get_broadcast_repos_dict(
+    mocker,
+    user_filesystem,
+    url_to_repo_info,
+    input_type,
+    mock_cwd,
+    mock_home,
+    expected_output,
+):
+    available_mockers = {}
 
-    def mock_clone_repo(cwd, **kwargs):
-        # already a temporary directory created by
-        #   _get_broadcast_repos_dict
-        target_dir = cwd[-1]
-        with open(f"{target_dir}/groups.json", "w") as f:
-            f.write(group_json_content)
-        with open(f"{target_dir}/repos.json", "w") as f:
-            f.write(repos_json_content)
+    if input_type == "url":
 
-    clone_repo_mocker = mocker.patch(
-        "subprocess.run",
-        side_effect=mock_clone_repo,
-    )
-    actual_group_dict, actual_repos_dict = _get_broadcast_repos_dict(
-        url_to_repo_info=url_to_repo_info
+        def mock_clone_repo(cwd, **kwargs):
+            # already a temporary directory created by
+            #   _get_broadcast_repos_dict
+            source_dir = user_filesystem / "repo_info_dir_json"
+            group_json_content = (source_dir / "groups.json").read_text()
+            repos_json_content = (source_dir / "repos.json").read_text()
+            target_dir = cwd[-1]
+            with open(f"{target_dir}/groups.json", "w") as f:
+                f.write(group_json_content)
+            with open(f"{target_dir}/repos.json", "w") as f:
+                f.write(repos_json_content)
+
+        mock_clone_repo = mocker.patch(
+            "subprocess.run",
+            side_effect=mock_clone_repo,
+        )
+    if mock_cwd:
+        available_mockers["mocker_cwd"] = mocker.patch(
+            "pathlib.Path.cwd",
+            return_value=user_filesystem / mock_cwd,
+        )
+    if mock_home:
+        available_mockers["mocker_home"] = mocker.patch(
+            "pathlib.Path.home",
+            return_value=user_filesystem / mock_home,
+        )
+    default_expected_group_dict, default_expected_repos_dict = (
```

Comment:
I think these should be in `pytest.mark.paramaetrize`.  Is there a reason you are not using our group standard here and putting the cases in there, minimizing the logic in the test itsefl?